armory/scenarios/poison.py

```diff
@@ -283,33 +321,112 @@ def load_dataset(self, eval_split_default="test"):
+        # )
+        # log.info(
+        #    f"Accuracy on benign test data--source class: {self.accuracy_on_benign_data_source_class.mean():.2%}"
+        # )
+        # log.info(
+        #    f"Accuracy on poisoned data--all classes: {self.accuracy_on_poisoned_data_all_classes.mean():.2%}"
+        # )
+        # log.info(
+        #    f"Attack success rate: {self.attack_success_rate.mean():.2%}"
+        #    )  # percent of poisoned source examples that get classified as target
+
         if self.use_poison:
-            self.accuracy_on_poisoned_data_all_classes = metrics.MetricList(
-                "categorical_accuracy"
+            self.hub.connect_meter(
+                Meter(
+                    "sample_accuracy_on_poisoned_data_all_classes",
+                    metrics.get_supported_metric("categorical_accuracy"),
+                    "scenario.y[benign]",
+                    "scenario.y_pred_adv[adversarial]",
+                    final=np.mean,
+                    final_name="accuracy_on_poisoned_data_all_classes",
+                    record_final_only=True,
+                )
+            )
+            # counts number of source images classified as target
+            self.hub.connect_meter(
+                Meter(
+                    "sample_attack_success_rate",
+                    metrics.get_supported_metric("categorical_accuracy"),
+                    "scenario.target_class_source",
+                    "scenario.y_pred_adv_source",
+                    final=np.mean,
+                    final_name="attack_success_rate",
+                    record_final_only=True,
+                )
             )
-            self.attack_success_rate = metrics.MetricList("categorical_accuracy")
             # attack_success_rate is not just 1 - (accuracy on poisoned source class)
             # because it only counts examples misclassified as target, and no others.
 
-        self.benign_test_accuracy_per_class = (
-            {}
-        )  # store accuracy results for each class
+        # TODO: write GlobalMeter ???
+        per_class_accuracy = metrics.get_supported_metric("per_class_accuracy")
+        self.hub.connect_meter(
+            Meter(
+                "sample_benign_test_accuracy_per_class",
+                metrics.get_supported_metric("identity_unzip"),
+                "scenario.y",
+                "scenario.y_pred",
+                final=lambda x: per_class_accuracy(*metrics.identity_zip(x)),
+                final_name="unpoisoned_test_accuracy_per_class",
+                record_final_only=True,
+            )
+        )
+
         if self.config["adhoc"].get("compute_fairness_metrics", False):
             self.fairness_metrics = FairnessMetrics(
                 self.config["adhoc"], self.use_filtering_defense, self
             )
+            if not self.check_run:
+                self.fairness_metrics.add_filter_perplexity()
```

Comment:
Adding filter_perplexity should be conditional on self.use_filtering_defense

Reply:
Done.